File: MyNotesProject/MyNotes/src/main/java/com/fsk/mynotes/services/DeleteNoteService.java
package com.fsk.mynotes.services;


import android.app.IntentService;
import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;

import com.fsk.common.database.DatabaseHelper;
import com.fsk.mynotes.constants.NoteExtraKeys;
import com.fsk.mynotes.data.Note;

/**
 * A service that deletes a note from the persistent storage
 */
public class DeleteNoteService extends IntentService {

    /**
     * Start the service for the specified note.
     *
     * @param context
     *         The context to use for starting the service.
     * @param note
     *         The note to delete.
     */
    public static void startService(@NonNull Context context, @NonNull Note note) {
        Intent intent = new Intent(context, DeleteNoteService.class);
        intent.putExtra(NoteExtraKeys.NOTE_KEY, note);
        context.startService(intent);
    }


    /**
     * Default Constructor.
     */
    public DeleteNoteService() {
        super(DeleteNoteService.class.getName());
    }


    @Override
    protected void onHandleIntent(final Intent intent) {
        Note note = intent.getParcelableExtra(NoteExtraKeys.NOTE_KEY);
        note.delete(DatabaseHelper.getDatabase());
    }
}

File: MyNotesProject/MyNotes/src/main/java/com/fsk/mynotes/services/DeleteNoteService.kt
package com.fsk.mynotes.services;


import android.app.IntentService
import android.content.Context
import android.content.Intent
import com.fsk.mynotes.constants.NOTE_KEY_EXTRA
import com.fsk.mynotes.data.Note

/**
 * Start the service for the specified note.
 *
 * @param note
 *         The note to delete.
 */
fun Context.startDeleteNoteService(note: Note) {

    val intent = Intent(this, DeleteNoteService::class.java).apply {
        putExtra(NOTE_KEY_EXTRA, note)
    }
    startService(intent);
}

/**
 * A service that deletes a note from the persistent storage
 */
class DeleteNoteService() : IntentService(DeleteNoteService::class.java.simpleName) {


    override fun onHandleIntent(intent: Intent?) {
        if (intent != null) {
            val note: Note = intent.getParcelableExtra(NOTE_KEY_EXTRA);
            note.delete(contentResolver);
        }
    }
}
